support-frontend/assets/pages/digital-subscription-checkout/components/datePicker/datePicker.jsx

```diff
@@ -0,0 +1,256 @@
+// @flow
+
+import React, { Component } from 'react';
+import DayPicker, { DateUtils } from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import { css } from '@emotion/core';
+import { space } from '@guardian/src-foundations';
+import { formatMachineDate } from 'helpers/dateConversions';
+import calendarIcon from './calendarIcon.png';
+import { type Option } from 'helpers/types/option';
+import { monthText } from 'pages/paper-subscription-checkout/helpers/subsCardDays';
+import { Input } from 'components/forms/input';
+import { withLabel } from 'hocs/withLabel';
+import { ThemeProvider } from 'emotion-theming';
+import { Button, buttonDefault } from '@guardian/src-button';
+import { Error } from 'components/forms/customFields/error';
+
+import './styles.scss';
+
+const calendarIconContainer = css`
+  padding: 0;
+  border: none;
+  width: 35px;
+  height: 35px;
+  align-self: flex-end;
+  margin: 0 0 ${space[1]}px ${space[4]}px ;
+`;
+
+const iconImage = css`
+  width: 100%;
+  height: 100%;
+`;
+
+const startDateGroup = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+const startDateFields = css`
+  display: inline-flex;
+`;
+
+const inputLayout = css`
+  width: 25%;
+`;
+
+const inputLayoutWithMargin = css`
+  width: 25%;
+  margin-right: ${space[3]}px;
+`;
+
+const validationButton = css`
+  margin-top: ${space[5]}px;
+`;
+
+const marginTop = css`
+  margin-top: ${space[5]}px;
+`;
+
+type PropTypes = {
+  // eslint-disable-next-line react/no-unused-prop-types
+  value: string | null,
+  onChange: Function,
+  // eslint-disable-next-line react/no-unused-prop-types
```

Comment:
Why do we need to disable eslint here?

Reply:
The first one (on line 61) we don't actually need, and I have removed it (thanks!). The second one I have kept because otherwise flow complains that I am passing the id prop into the withError hoc but it's not declared in PropTypes for this component. And since it isn't actually used in this component, I'm not clear how to overcome the issue other than by disabling eslint for this line. Very happy to hear if there is a better way of handling this!

Reply:
Doesn't this mean that we actually need to use the id somewhere within this component to give the error message something to refer to? Otherwise the id set on this component will not make its way into the markup.

Reply:
I thought it was consumed by the withError hoc. Could be totally misunderstanding though!

Reply:
Thanks for looking into this and fixing it @rupertbates